keras_core/layers/preprocessing/random_crop.py

```diff
@@ -75,7 +75,33 @@ def __init__(self, height, width, seed=None, name=None, **kwargs):
     def call(self, inputs, training=True):
         if not isinstance(inputs, (tf.Tensor, np.ndarray, list, tuple)):
             inputs = tf.convert_to_tensor(backend.convert_to_numpy(inputs))
-        outputs = self.layer.call(inputs, training=training)
+
+        input_shape = tf.shape(inputs)
+        h_diff = input_shape[H_AXIS] - self.height
+        w_diff = input_shape[W_AXIS] - self.width
+
+        def random_crop():
+            dtype = input_shape.dtype
+            rands = tf.random.uniform([2], 0, dtype.max, dtype)
+            h_start = rands[0] % (h_diff + 1)
+            w_start = rands[1] % (w_diff + 1)
+            return tf.image.crop_to_bounding_box(
```

Comment:
Could we do this in a backend-agnostic way? I think `inputs[]` should work?

Reply:
@fchollet 
Do you recommend something like [this](https://github.com/keras-team/keras-io/blob/master/examples/vision/mirnet.py#L76-L92)?

Reply:
Sounds fine! But make sure to use `keras.random` ops seeded with a seed generator and make sure it supports both channel formats.